api/src/main/java/org/apache/iceberg/transforms/PartitionSpecVisitor.java

```diff
@@ -55,17 +56,24 @@
       } else if (transform instanceof Truncate) {
         results.add(visitor.truncate(sourceName, field.sourceId(),
             ((Truncate<?>) transform).width()));
-      } else if (transform == Dates.YEAR || transform == Timestamps.YEAR) {
+      } else if (transform == Dates.YEAR || isTimestamp(transform, ChronoUnit.YEARS)) {
```

Comment:
Will users of the visitor not need to know the zone offset?

Reply:
So far I don't have a good idea about this. I just modify the reference to timestamps accordingly.

Reply:
After modification,`timeZone` is already `final`.
Is there any validation work that needs to be done here?